web/modules/custom/rdf_entity/src/Plugin/Field/FieldWidget/UrlFileSynchronizerWidget.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Drupal\rdf_entity\Plugin\Field\FieldWidget;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\file\Entity\File;
+use Drupal\link\Plugin\Field\FieldWidget\LinkWidget;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'url_file_synchronizer_widget' widget.
```

Comment:
The issue here is that the entity has two fields (relative to this widget). A file field and a URI field.
The user will be able to provide files for this distribution either by URL or by directly uploading the file to the website.
Now, the thing is that the URL field is part of the rdf entity and the file field is not (it is just a helper for the user to be able to upload the files). So in order to keep the rdf entity up to date, we need to have the URLs passed to the URL field. This implies that if a user has filled both fields, the URL field must be updated (same applies if a link or file is removed during an edit).
After discussions, we thought that it would be better to do this formally in a widget than handling it in a hook.

I thought that since we are going to contribute this module, someone else might need something like this so I thought that it should remain to the base entity (plus for reusability reasons, so that we can avoid dependencies on the distribution entitiy).